docs/library/pyb.Flash.rst

```diff
@@ -0,0 +1,52 @@
+.. currentmodule:: pyb
+.. _pyb.Flash:
+
+class Flash -- access to built-in flash storage
+===============================================
+
+The Flash class allows direct access to the primary flash device on the pyboard.
+
+In most cases, to store persistent data on the device, you'll want to use a
+higher-level abstraction, for example the filesystem via Python's standard file
+API, but this interface is useful to :ref:`customise the filesystem
+configuration <filesystem>` or implement a low-level storage system for your
+application.
+
+Constructors
+------------
+
+.. class:: pyb.Flash()
+
+   Create and return a block device that represents the flash device presented
+   to the USB mass storage interface.
+
+   It includes a virtual partition table at the start, and the actual flash
+   starts at block ``0x100``.
+
+   This constructor is deprecated and will be removed in a future version of MicroPython.
+
+.. class:: pyb.Flash(start, [len])
```

Comment:
I think this should be: `pyb.Flash(*, start=-1, len=-1)` (note that `*` should be entered in rst as `\*`).  Because they args are kw-only, and if you specify -1 then that is actually the default value and it'll auto-detect.

Reply:
Done